CRM/Financial/BAO/FinancialItem.php

```diff
@@ -109,7 +109,9 @@ public static function add($lineItem, $contribution, $taxTrxnID = FALSE, $trxnId
     }
     else {
       $accountRelName = 'Income Account is';
-      if (property_exists($contribution, 'revenue_recognition_date') && !CRM_Utils_System::isNull($contribution->revenue_recognition_date)) {
+      if (property_exists($contribution, 'revenue_recognition_date') &&
```

Comment:
this is a code smell. If it's necessary to fix an e-notice for tests sake I think we should define revenue_recognition_date as a BAO property (if not we can ignore for now :-)